feat(geoipupdater) add a rollout restart and update only if necessary

smerle33 opened the pull request:
ref https://github.com/jenkins-infra/helpdesk/issues/4441

following https://github.com/jenkins-infra/helpdesk/issues/4278 and https://github.com/jenkins-infra/helpdesk/issues/4240

we had to improve the update process with only if needed pattern and a rollout restart included.

dduportal reviewed (dismissed):
This change uses the image 0.3.3 while it seems 0.3.5 is required: https://github.com/jenkins-infra/docker-geoipupdate/releases/tag/0.3.5

smerle33 commented:
I corrected the secrets for the production fileshare here https://github.com/jenkins-infra/charts-secrets/commit/86439c0050f23fc1c3cd9eef3e4ac14b842dd29a

and added a staging secrets file, that we will remove, with the terraform for staging when everything will be ok 
TODO: cleanup
https://github.com/jenkins-infra/azure/blob/0dd69e354c3d764d72763ae3f4b23ec29b67cf69/publick8s.tf#L371-L388

dduportal reviewed (approved):
- We are watching 2 pages on both get.jenkins.io and mirrors.updates.jenkins.io (with the ?mirrorlist query string)
- We are also watching pods on the 3 namespaces

Let's roll!
